packages/babel-generator/src/generators/types.js

```diff
@@ -120,7 +120,11 @@ export function NullLiteral() {
 
 export function NumericLiteral(node: Object) {
   const raw = this.getPossibleRaw(node);
-  const value = node.value + "";
+  const opts = this.format.jsescOption;
+  if (this.format.jsonCompatibleStrings) {
+    opts.json = true;
+  }
+  const value = jsesc(node.value, opts);
```

Comment:
Can you add tests for these two `jsesc` options? One should expect both `numbers` and `lowercasehex` should work for `NumericLiteral`.

Reply:
Ok:+1:, do I also need to set
```
{
    jsescOption: {
        numbers: 'hexadecimal'
    }
}
```
in the tests?

Reply:
Yep that is exactly what #10960 is requiring.